output/clickhouse_output.go

```diff
@@ -21,6 +23,17 @@ const (
 	CLICKHOUSE_DEFAULT_FLUSH_INTERVAL = 30
 )
 
+var transIntColumn = make(map[string]string)
```

Comment:
应该是 []string 类型吧。 map 类型没用上。

Reply:
后面有用到map的key是字段名，v是字段类型
```
for key1, value1 := range c.desc {
			switch value1.Type {
			case "Int64", "UInt64", "Int32", "UInt32", "Int16", "UInt16", "Int8", "UInt8", "Nullable(Int64)", "Nullable(Int32)", "Nullable(Int16)", "Nullable(Int8)":
```suggestion
var transIntColumn = make(map[string]string)
```
				transIntColumn[key1] = value1.Type
			case "Array(String)", "Array(Int64)", "Array(Int32)", "Array(Int16)", "Array(Int8)":
				transArrayColumn[key1] = value1.Type
			case "Float64", "Float32":
				transFloatColumn[key1] = value1.Type
			}
			c.fields = append(c.fields, key1)
		}
		transIntColumnSize = len(transIntColumn)
		transArrayColumnSize = len(transArrayColumn)
		transFloatColumnSize = len(transFloatColumn)
		return
```

Reply:
`v是字段类型` ，但字段类型没有用到吧

Reply:
数组处理这块有用到v 要根据columnArrayType 判断
```
if transArrayColumnSize > 0 {
				for keyArray, columnArrayType := range transArrayColumn {
					keyArrayValue := event[keyArray]
					if keyArrayValue == nil {
						continue
					}
					switch columnArrayType {
					case "Array(Int64)", "Array(Int32)", "Array(Int16)", "Array(Int8)":
						arrayIntValue := keyArrayValue.([]interface{})
						arrayIntLen := len(arrayIntValue)
						ints := make([]int, arrayIntLen)
						for i := 0; i < arrayIntLen; i++ {
							arrayIntConverterValue, err := c.intConverter(arrayIntValue[i])
							if err == nil {
								ints[i] = arrayIntConverterValue.(int)
							} else {
								glog.V(10).Infof("ch_output convert arrayIntType error: %s", err)
								ints[i] = 0
							}
						}
						event[keyArray] = ints
					}
				}
			}
```